cirq/ops/eigen_gate.py

```diff
@@ -64,23 +82,57 @@ def _with_exponent(self: TSelf,
         pass
 
     @abc.abstractmethod
-    def _eigen_components(self) -> List[Tuple[float, np.ndarray]]:
-        """A decomposition of the gate into (λ_half_turns, Σ|λ⟩⟨λ|) pieces.
+    def _eigen_components(self) -> List[Union[EigenComponent,
+                                              Tuple[float, np.ndarray]]]:
+        """Describes the eigendecomposition of the gate's matrix.
 
         Returns:
-            A list of tuples. Each tuple corresponds to an eigenspace of the
-            gate. The first component of a tuple is how much that eigenspace
-            should be phased by applying the gate, in half_turn units.
-            The second component is the projection of the gate's matrix into
-            that eigenspace.
-
-            For example, the Pauli X gate's eigen components method would
-            return [
-                (0, np.array([[0.5, 0.5],
-                              [0.5, 0.5]])),
-                (1, np.array([[+0.5, -0.5],
-                              [-0.5, +0.5]])),
-            ].
+            A list of EigenComponent tuples. Each tuple in the list
+            corresponds to one of the eigenspaces of the gate's matrix. Each
+            tuple has two elements. The first element of a tuple is the θ in
+            λ = exp(i π θ) (where λ is the eigenvalue of the eigenspace). The
+            second element is a projection matrix onto the eigenspace.
+
+        Examples:
+            The Pauli Z gate's eigencomponents are:
+
+                [
+                    (0, np.array([[1, 0],
+                                  [0, 0]])),
+                    (1, np.array([[0, 0],
+                                  [0, 1]])),
+                ]
+
+            Valid eigencomponents for Rz(π) = iZ are:
```

Comment:
Rz(π) = exp(-i π Z / 2) is actually -iZ.

Reply:
Oops. Fixed.